apps/sms/test/marionette/conversation_test.js

```diff
@@ -314,4 +314,11 @@ marionette('Conversation Panel Tests', function() {
       assert.equal(downloadedMessage.content.trim(), 'Attachment');
     });
   });
+
+  suite('Edit Mode', function() {
+    messagesApp.launch();
```

Comment:
Okay, you've started, let's slowly learn new things and move forward, step-by-step :)

* 1st step: you've created ```suite``` - ```suite``` is a bunch of tests, suite can contain ```setup```, ```teardown``` and ```test``` inside - I'd recommend you to read about first to understand what code you need to put in ```setup```, what in ```test``` and so on, e.g. you can read [1] and [2], and of course there is a plenty of information in the internet.

* 2nd step: assuming you finished 1st step and you know how ```suite``` should look like: so as you know we can't run Messages integration tests on the real phone - otherwise we should send real SMS/MMS - so it's hard, unreliable and costly for the tests. At the same time in tests we need some test data already (like in your case, we need conversation with few messages to select them). For that purpose we "mock/fake" data and different Web API that are not available in test environment (e.g. see [3] for the mock of [mozMobileMessage](https://developer.mozilla.org/en-US/docs/Web/API/MozMobileMessageManager) API). To fake data we use [4]. So in your case you need have _one_ conversation with few messages in it, and you should create this fake data (please keep in mind that data you see when you run app in browser is totally different from one you fake in integration tests). We do that a lot, see example of that in [5]. In [5] we use special ```ThreadGenerator``` that accepts parameter of the messages to generate and generate fake conversation with messages, then we run Messages app (via ```messagesApp.launch();```) and only then we can say Messages app that we'd like it to have our fake data (via ```storage.setMessagesStorage([thread], ThreadGenerator.uniqueMessageId);```).

* 3rd step: once we loaded app, faked data we can create test "Inbox" view and use it to manage real Inbox view [6].

It's not that difficult as it seems, but once you understand and complete all these steps you'll have quite powerful knowledge about tests in general and Messages app integration tests in particular. So now please take your time to learn all this and ping me (you can do this here, I follow GitHub comments) if you have any questions that you can't find answers for :)

Thanks!



[1] http://code.tutsplus.com/articles/tdd-terminology-simplified--net-30626
[2] http://dailyjs.com/2011/12/08/mocha/
[3] https://github.com/mozilla-b2g/gaia/blob/master/apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js
[4] https://github.com/mozilla-b2g/gaia/blob/master/apps/sms/test/marionette/lib/storage.js
[5] https://github.com/mozilla-b2g/gaia/blob/4973f57cd8f9a62a95f783a24eac32da2bde99fc/apps/sms/test/marionette/conversation_test.js#L273-L286
[6] https://github.com/mozilla-b2g/gaia/blob/4973f57cd8f9a62a95f783a24eac32da2bde99fc/apps/sms/test/marionette/conversation_test.js#L289-L291